pkg/ksym/ksym.go

```diff
@@ -45,29 +49,34 @@ type Cache struct {
 	lastCacheInvalidation time.Time
 	updateDuration        time.Duration
 	fastCache             map[uint64]string
-	Stats                 CacheStats
+	metrics               *metrics
 	mtx                   *sync.RWMutex
 }
 
 type realfs struct{}
 
 func (f *realfs) Open(name string) (fs.File, error) { return os.Open(name) }
 
-func (c CacheStats) HitRate() float64 {
-	return 100 * float64(c.Hits) / float64(c.Total)
-}
-
-func (c CacheStats) String() string {
-	return fmt.Sprintf("Ksym hit rate: %.2f%% (%d Total cache accesses)", c.HitRate(), c.Total)
+func newMetrics(reg prometheus.Registerer, target model.LabelSet) *metrics {
+	return &metrics{
+		ksymCacheHitRate: promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Name:        "parca_agent_profiler_kernel_symbolizer_cache_total",
+				Help:        "Stats for the kernel symbolizer cache",
+				ConstLabels: map[string]string{"target": target.String()},
+			},
+			[]string{"type"},
+		),
+	}
 }
 
-func NewKsymCache(logger log.Logger) *Cache {
+func NewKsymCache(logger log.Logger, reg prometheus.Registerer, target model.LabelSet) *Cache {
```

Comment:
I think this is the perfect use case for https://github.com/prometheus/client_golang/blob/0222f88f4a941059843b49d9f56b42b9fea90d97/prometheus/wrap.go#L47-L52

Rather than passing target labels just wrap the register and pass it.